lib/proxy_api/resource.rb

```diff
@@ -1,26 +1,18 @@
+require 'securerandom'
+
 module ProxyAPI
   class Resource
     attr_reader :url
 
     def initialize(args)
       raise("Must provide a protocol and host when initialising a smart-proxy connection") unless (url =~ /^http/)
 
-      @connect_params = {:timeout => Setting[:proxy_request_timeout], :open_timeout => 10, :headers => { :accept => :json },
-                        :user => args[:user], :password => args[:password]}
+      @connect_params = {:timeout => Setting[:proxy_request_timeout], :open_timeout => 10,
+                         :headers => { :accept => :json, :x_request_id => request_id },
+                         :user => args[:user], :password => args[:password]}
 
       # We authenticate only if we are using SSL
-      if url.match(/^https/i)
-        cert         = Setting[:ssl_certificate]
-        ca_cert      = Setting[:ssl_ca_file]
-        hostprivkey  = Setting[:ssl_priv_key]
-
-        @connect_params.merge!(
-          :ssl_client_cert  =>  OpenSSL::X509::Certificate.new(File.read(cert)),
-          :ssl_client_key   =>  OpenSSL::PKey::RSA.new(File.read(hostprivkey)),
-          :ssl_ca_file      =>  ca_cert,
-          :verify_ssl       =>  OpenSSL::SSL::VERIFY_PEER
-        ) unless Rails.env == "test"
-      end
+      @connect_params.merge!(ssl_auth_params) if url.match(/\Ahttps/i) && !Rails.env.test?
```

Comment:
Refactored into a separate method as CodeClimate warned about overall increase in complexity of `initialize`.